src/ContentHubClient.php

```diff
@@ -1199,4 +1202,40 @@ protected function addRequestResponseHandler(array $config): void
 
         $config['handler']->push(new RequestResponseHandler($this->logger));
     }
+
+    /**
+     * Appends search criteria header.
+     *
+     * @param array $args
+     *   Method arguments.
+     *
+     * @return mixed
+     */
+    protected function addSearchCriteriaHeader($args)
+    {
+        $uri = isset($args[0]) ? $args[0] : '';
+        $options = isset($args[1]) ? $args[1] : [];
+
+        if (strpos($uri, '?') === false) {
+            return $args;
+        }
+
+        list($baseUri, $queryString) = explode('?', $uri);
+        if (empty($queryString)) {
+            return $args;
+        }
+
+        parse_str($queryString, $parsedQueryString);
+        $messageBody = isset($options['body']) ? $options['body'] : '';
+
+        $criteria = SearchCriteriaBuilder::createFromArray($parsedQueryString);
+        $criteria = json_encode($criteria);
```

Comment:
I would merge these two into one, since the result of the first assignment is only used in the  immediate statement.